src/aleph/http/client.clj

```diff
@@ -100,6 +100,21 @@
           nil))
       (no-url req))))
 
+(defn exception-handler [ctx ex response-stream]
+  (cond
+    ;; could happens when io.netty.handler.codec.http.HttpObjectAggregator
+    ;; is part of the pipeline
+    (instance? TooLongFrameException ex)
+    (s/put! response-stream ex)
+
+    ;; when SSL handshake failed
+    (netty/ssl-handshake-error? ex)
+    (let [^Throwable handshake-error (.getCause ^Throwable ex)]
+      (s/put! response-stream handshake-error))
+
+    (not (instance? IOException ex))
+    (log/warn ex "error in HTTP client")))
```

Comment:
On the error-handling branch, Oleksii use to close the channel.
Maybe we should do it on this PR [1]

[1] : https://github.com/clj-commons/aleph/blob/error-handling-1.0.0/src/aleph/http/client.clj#L122

Reply:
I def agree we should be doing more than logging a warning. However, I'm not sure we should do as Oleksii did, and send the exception to the client, at least not by default. It's handy for debugging, but a bad idea for security.

Reply:
@DerGuteMoritz If you want to close the channel, I can wait. Otherwise, is there anything else to this? It's ready, right?

Reply:
Hmm yeah closing the channel seems reasonable. I wonder if this means that the current code leaks connections on exception? In any case, will push a change to that effect since it certainly shouldn't hurt either.

> However, I'm not sure we should do as Oleksii did, and send the exception to the client, at least not by default. It's handy for debugging, but a bad idea for security.

Note that the code in question here is the HTTP *client* in which case I think forwarding the exception to the caller is fine? Your point is certainly true wrt the HTTP *server*'s default exception handler which indeed echoes back exceptions verbatim with no way to disable this. In fact, [there is another open PR addressing exactly this issue](https://github.com/clj-commons/aleph/pull/521).

Reply:
Pushed! Should be good to go then... :crossed_fingers: Another sanity-check welcome, of course.

Reply:
Yes, you're right, the _client_ should get exceptions, the _server_ should not send them.

The connections should be a dynamic pool. If too many are closed, more will be created as necessary, up to the limits.